metagraph/src/annotation/binary_matrix/row_diff/row_diff.hpp

```diff
@@ -148,6 +152,70 @@ BinaryMatrix::SetBitPositions RowDiff<BaseMatrix>::get_row(Row row) const {
     return result;
 }
 
+template <class BaseMatrix>
+std::vector<BinaryMatrix::SetBitPositions>
+RowDiff<BaseMatrix>::get_rows(const std::vector<Row> &row_ids) const {
+    // diff rows annotating nodes along the row-diff paths
+    std::vector<Row> rd_ids;
+    rd_ids.reserve(row_ids.size() * 2);
+    // map row index to its index in |rd_rows|
+    VectorMap<Row, size_t> node_to_rd;
+    node_to_rd.reserve(row_ids.size() * 2);
+    // row-diff paths, indexes to |rd_rows|. The last index always points to
+    // an anchor or to a row which had been reconstructed and is stored in full.
+    std::vector<std::vector<size_t>> rd_paths(row_ids.size());
+
+    const graph::boss::BOSS &boss = graph_->get_boss();
+
+    for (size_t i = 0; i < row_ids.size(); ++i) {
+        Row row = row_ids[i];
+
+        graph::boss::BOSS::edge_index boss_edge = graph_->kmer_to_boss_index(
+                graph::AnnotatedSequenceGraph::anno_to_graph_index(row));
+
+        while (true) {
+            row = graph::AnnotatedSequenceGraph::graph_to_anno_index(
+                    graph_->boss_to_kmer_index(boss_edge));
+
+            auto [it, is_new] = node_to_rd.try_emplace(row, rd_ids.size());
+            rd_paths[i].push_back(it.value());
```

Comment:
```suggestion
            rd_paths[i].push_back(*it);
```
you only need `it.value()` to access a non-const reference. for other cases, dereferencing should be good enough

Reply:
Shouldn't it be `it->second`?
For `*it`, it is not quite clear what it returns, could return a pair.

Reply:
yes, true. not that important anyway